api/v1alpha1/openstackimage_types.go

```diff
@@ -63,16 +51,21 @@ type OpenStackImageSpec struct {
 	// image.
 	MinRAM int `json:"minRam,omitempty"`
 
+	// Name of the OpenStack resource.
+	Name string `json:"name,omitempty"`
+
 	// protected is whether the image is not deletable.
-	Protected *bool `json:"protected,omitempty"`
+	// +kubebuilder:default:=false
+	Protected bool `json:"protected,omitempty"`
 
-	// properties is a set of properties, if any, that are associated with
-	// the image.
-	// Properties map[string]string `json:"-"`
+	// Tags is a set of image tags.
+	// Each tag is a string of at most 255 chars.
+	Tags []string `json:"tags,omitempty"`
 
-	// Source defines the content of the image. If not set, the image is
-	// left empty.
-	Source *OpenStackImageSource `json:"source,omitempty"`
+	// Visibility defines who can see/use the image.
+	// +kubebuilder:validation:Enum:="default";"public";"private";"shared";"community"
+	// +kubebuilder:default:="default"
+	Visibility string `json:"visibility,omitempty"`
```

Comment:
Consistently with our new policies, `Visibility` should be `*string`, so that when the user lets it blank, we just don't send the key to the Glance API. Glance will set a default
The validation stays in place, but `kubebuilder:default` goes away.

```suggestion
	// Visibility defines who can see/use the image.
	// +kubebuilder:validation:Enum:="public";"private";"shared";"community"
	Visibility *string `json:"visibility,omitempty"`
```